src/linter/root.go

```diff
@@ -950,11 +945,17 @@ func (d *rootWalker) enterClassMethod(meth *ir.ClassMethodStmt) bool {
 	}
 	doc := d.parsePHPDoc(meth.MethodName, meth.Doc, meth.Params)
 	d.reportPhpdocErrors(meth.MethodName, doc.errs)
-	phpdocReturnType := doc.returnType
+	phpDocReturnType := doc.returnType
 	phpDocParamTypes := doc.types
 
 	class := d.getClass()
-	params, minParamsCnt := d.parseFuncArgs(meth.Params, phpDocParamTypes, sc, nil)
+
+	hintReturnType, has := d.parseTypeNode(meth.ReturnType)
+	if has && !doc.inherit {
```

Comment:
`has` is not a good variable name.
99% of time it should be just `ok` or, if you don't want to pollute the scope:
```go
if hintReturnType, ok := d.parseTypeNode(meth.ReturnType); ok {
  // Use hintReturnType
}
```

Reply:
Done.